include/motis/gbfs/diff.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <functional>
+
+namespace motis::gbfs {
+
+template <typename Collection1, typename Collection2>
```

Comment:
`utl::sorted_diff`?

Please never use `std::function` - it's too heavy weight at compile time, too slow at runtime and in general too complex with all its magic and type erasure.

Especially in headers just use a template parameter or `auto&&`. If you want to express type requirements, C++20 concepts are useful.